worlds/messenger/__init__.py

```diff
@@ -401,19 +397,21 @@ def remove(self, state: "CollectionState", item: "Item") -> bool:
             state.prog_items[self.player]["Shards"] -= int(item.name.strip("Time Shard ()"))
         return change
 
-
-def generate_output(world: MessengerWorld, output_directory: str) -> None:
-    out_path = output_path(world.multiworld.get_out_file_name_base(1) + ".aptm")
-    if "The Messenger\\Archipelago\\output" not in out_path:
-        return
-    import orjson
-    data = {
-        "name": world.multiworld.get_player_name(world.player),
-        "slot_data": world.fill_slot_data(),
-        "loc_data": {loc.address: {loc.item.name: [loc.item.code, loc.item.flags]}
-                     for loc in world.multiworld.get_filled_locations() if loc.address},
-    }
-
-    output = orjson.dumps(data, option=orjson.OPT_NON_STR_KEYS)
-    with open(out_path, "wb") as f:
-        f.write(output)
+    @classmethod
+    def stage_generate_output(cls, multiworld: MultiWorld, output_directory: str) -> None:
```

Comment:
You forgot to add a comment explaining what this is. it's super confusing if I didn't know already.